Source/Core/VideoBackends/D3D/PixelShaderCache.cpp

```diff
@@ -146,7 +146,7 @@ const char depth_matrix_program[] = {
 	" in float4 pos : SV_Position,\n"
 	" in float3 uv0 : TEXCOORD0){\n"
 	"	float4 texcol = Tex0.Sample(samp0,uv0);\n"
-	"	int depth = int(round(texcol.x * float(0xFFFFFF)));\n"
+	"	int depth = int(round(texcol.x * 16777216.0));\n"
```

Comment:
After multiplying with 16777216.0, you need to make sure that the result is always smaller than 16777216.0. I think round() is also wrong now, floor sounds better to me.